### PR TITLE
Site settings UI

### PR DESCRIPTION
**Pull request by sebastienros**

### CODE REVIEW

File: src/Orchard.SiteSettings.Core/SiteSettingsExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
+﻿using Newtonsoft.Json.Linq;
+
+namespace Orchard.Settings
+{
+    public static class SiteSettingsExtensions
+    {
+        /// <summary>
+        /// Extracts the specified type of porperty.
```
Comment (agriffard) [inline src/Orchard.SiteSettings.Core/SiteSettingsExtensions.cs]: porperty => property

File: src/Orchard.SiteSettings.Core/SiteSettingsExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        /// <typeparam name="T">The type of the property to extract.</typeparam>
+        /// <returns>The default value of the requested type if the property was not found.</returns>
+        public static T As<T>(this ISite site)
+        {
+            var typeName = typeof(T).Name;
+            return site.As<T>(typeName);
+        }
+
+        /// <summary>
+        /// Extracts the specified named porperty.
```
Comment (agriffard) [inline src/Orchard.SiteSettings.Core/SiteSettingsExtensions.cs]: porperty => property

File: src/Orchard.Cms.Web/Modules/Lucene/Startup.cs [hunk 3]
```diff
@@ -24,7 +28,8 @@ public override void ConfigureServices(IServiceCollection services)
             services.AddScoped<IContentPartFieldDefinitionDisplayDriver, ContentPartFieldIndexSettingsDisplayDriver>();
             services.AddScoped<INavigationProvider, AdminMenu>();
```
Comment (agriffard) [inline src/Orchard.Cms.Web/Modules/Lucene/Startup.cs]: Missing services.AddScoped<IPermissionProvider, Permissions>();

Comment (sebastienros) [inline src/Orchard.Cms.Web/Modules/Lucene/Startup.cs]: I fixed it for this module. I have done it on other modules as it's not part of this PR and we have your bug to track it.

**Comment by sebastienros** [comment]
I am pushing another commit then I will merge.
